src/components/tile-manager/tile.ts

```diff
@@ -402,14 +402,40 @@ export default class IgcTileComponent extends EventEmitterMixin<
   private _handleDragOver(parameters: DragCallbackParameters) {
     const match = parameters.state.element as IgcTileComponent;
     const { clientX, clientY } = parameters.event;
+    const { left, top, width, height } = match.getBoundingClientRect();
 
-    if (this._dragStack.peek() === match) {
-      const { x, y } = getCenterPoint(match);
+    const relativeX = (clientX - left) / width;
+    const relativeY = (clientY - top) / height;
+
+    if (this._previousPointerPosition) {
+      const deltaX = clientX - this._previousPointerPosition.x;
+      const deltaY = clientY - this._previousPointerPosition.y;
+
+      const movingRight = deltaX > 0;
+      const movingLeft = deltaX < 0;
+      const movingDown = deltaY > 0;
+      const movingUp = deltaY < 0;
+
+      const isHorizontalMove = Math.abs(deltaX) > Math.abs(deltaY);
+      const isVerticalMove = Math.abs(deltaY) > Math.abs(deltaX);
```

Comment:
I think we should abstract this state into the drag controller itself. It can hold the previous pointer coordinates when a drag is initiated and return the _direction_ while dragging in its callback parameters. At least it will simplify the code inside the tile drag handler.